src/main/java/com/ays/auth/service/impl/AysInvalidTokenServiceImpl.java

```diff
@@ -0,0 +1,55 @@
+package com.ays.auth.service.impl;
+
+import com.ays.auth.model.entity.AysInvalidTokenEntity;
+import com.ays.auth.repository.AysInvalidTokenRepository;
+import com.ays.auth.service.AysInvalidTokenService;
+import com.ays.auth.util.exception.TokenAlreadyInvalidatedException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * AysInvalidTokenServiceImpl is an implementation of the AysInvalidTokenService interface that provides
+ * functionality for managing and checking the validity of tokens.
+ */
+@Service
+@RequiredArgsConstructor
```

Comment:
Since we are not only performing read operations but also handling saving operations within this service, adding `@Transactional(readOnly = true)` may not be the correct approach. Adding the `@Transactional` annotation would not serve any specific purpose for this service as there are no interdependent repository operations.